### PR TITLE
Add signIn with Google

### PR DESCRIPTION
**Pull request by jeanpimentel**

### CODE REVIEW

File: app/src/main/java/com/ciandt/dragonfly/example/features/login/LoginContract.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package com.ciandt.dragonfly.example.features.login
+
+import com.ciandt.dragonfly.example.shared.BasePresenterContract
+import com.google.android.gms.auth.api.signin.GoogleSignInResult
+import com.google.firebase.auth.FirebaseUser
+
+/**
+ * Created by iluz on 6/20/17.
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/login/LoginContract.kt:8]: :)

File: app/src/main/java/com/ciandt/dragonfly/example/features/login/LoginPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package com.ciandt.dragonfly.example.features.login
+
+import com.ciandt.dragonfly.example.infrastructure.DragonflyLogger
+import com.ciandt.dragonfly.example.shared.BasePresenter
+import com.google.android.gms.auth.api.signin.GoogleSignInResult
+
+/**
+ * Created by iluz on 6/20/17.
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/login/LoginPresenter.kt:8]: :)

**Review by ivamluz — APPROVED** [review]
Approved with comments.

Also, I didn't see any checking in the other activities to redirect to login activity in case the users is not signed in. Don't we need to add this?

**Comment by jeanpimentel** [comment]
It's in another task

**Pull request merged**